ical/iter.py

```diff
@@ -65,58 +65,30 @@ def __iter__(self) -> Iterator[T]:
         return RecurIterator(self._item_cb, iter(self._recur))
 
 
-class PeekingIterator(Iterator[T]):
-    """An iterator with a preview of the next item.
-
-    The primary purpose is to implement a merged iterator where it is needed to
-    see the next item in the iterator in order to decide which child iterator
-    to pull from.
-    """
-
-    def __init__(self, iterator: Iterator[T]):
-        """Initialize PeekingIterator."""
-        self._iterator = iterator
-        self._next = next(self._iterator, None)
-
-    def __iter__(self) -> Iterator[T]:
-        """Return this iterator."""
-        return self
-
-    def peek(self) -> T | None:
-        """Peek at the next item without consuming."""
-        return self._next
-
-    def __next__(self) -> T:
-        """Produce the next item from the merged set."""
-        result = self._next
-        self._next = next(self._iterator, None)
-        if result is None:
-            raise StopIteration()
-        return result
-
-
 class MergedIterator(Iterator[T]):
     """An iterator with a merged sorted view of the underlying sorted iterators."""
 
     def __init__(self, iters: list[Iterator[T]]):
         """Initialize MergedIterator."""
-        self._iters = [PeekingIterator(iterator) for iterator in iters]
+        self._heap: list[tuple[T, Iterator[T]]] = []
+        for iterator in iters:
+            next_item: T | None = next(iterator, None)
+            if next_item:
+                heapq.heappush(self._heap, (next_item, iterator))
```

Comment:
Maybe keep a separate copy of the list of iterators (with `tuple` maybe?), then enumerate the iterators and store the index of the iterator in the heap instead to break ties in a `heapq`-independent way.

Reply:
I was writing a test to exercise the other issue you pointed (basically values in the list that would evaluate truthiness of values in the iterator) and it exercised the issue you're pointing out here too. Basically, the heap was trying to compare the iterators and failed:
```
           TypeError: '<' not supported between instances of 'list_iterator' and 'list_iterator'
```
I've updated this to have the indexes into a list of iterators as you suggested.

Reply:
Oh yes it actually crashes if you do not use indices! Good catch.